staging/src/k8s.io/kubectl/pkg/drain/drain_test.go

```diff
@@ -220,9 +220,29 @@ func TestCheckEvictionSupport(t *testing.T) {
 }
 
 func TestDeleteOrEvict(t *testing.T) {
-	for _, evictionSupported := range []bool{true, false} {
-		evictionSupported := evictionSupported
-		t.Run(fmt.Sprintf("evictionSupported=%v", evictionSupported),
+	tests := []struct {
```

Comment:
Would it be better to turn this into map[string]struct{} so we can name each of the test cases? I think that could be useful to provide more context on what is being tested in each example. We can then use the name in the `t.Run` call on line 245.

Reply:
That's not really a pattern I've seen elsewhere.  We could do something like this: https://github.com/kubernetes/kubernetes/blob/366b834284da282cfe296c860f6a249b3b8efd89/staging/src/k8s.io/kubectl/pkg/drain/drain_test.go#L46

Honestly, I don't think it's really necessary in this case, there's not really much to describe and there are only 4 tests.

Reply:
Both patterns are used in the code base, I'm ok with either one. I just would like the name of the test used in `t.Run` to uniquely identify each case. It makes it much easier to understand what is going on if tests fail, in particular when it runs on Prow. Also, with a unique name we don't need to print the number of the test case in the `t.Errorf` statements.

Reply:
@mortent @seans3 I went ahead and added the 'description' to be consistent in the package.